### PR TITLE
Update out.md to add a usage example as well as limitations.

### PR DESCRIPTION
**Pull request by cjvandyk**
Update out.md to add a usage example as well as limitations.

## Summary

Added a usage example.
Added a list of limitations.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/csharp/language-reference/keywords/out.md](https://github.com/dotnet/docs/blob/4608c61c1c84b6271d201cff3550c895b086c1a8/docs/csharp/language-reference/keywords/out.md) | [out (C# Reference)](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/out?branch=pr-en-us-40020) |

### CODE REVIEW

File: docs/csharp/language-reference/keywords/out.md [hunk 1]
```diff
@@ -18,6 +18,36 @@ You can use the `out` keyword in two contexts:
 
 - In [generic type parameter declarations](out-generic-modifier.md) for interfaces and delegates, which specifies that a type parameter is covariant.
 
+The `out` keyword is especially useful when a method needs to return more than one value since more than one `out` parameter can be used e.g.
+
+```cs
```
Comment (BillWagner) [inline docs/csharp/language-reference/keywords/out.md:23]: Let's spell out csharp:

```suggestion
```csharp
```

File: docs/csharp/language-reference/keywords/out.md [hunk 2]
```diff
@@ -18,6 +18,36 @@ You can use the `out` keyword in two contexts:
...
+
+The following limitations apply to using the `out` keyword:
+
+- The `out` parameters are not allowed to be used in asynchronous methods.
+
+- The `out` parameters are not allowed to be used in iterator methods.
+
+- Properties cannot be passed as `out` parameters.
+
+
```
Comment (BillWagner) [inline docs/csharp/language-reference/keywords/out.md:50]: Let's remove extra blank lines:

```suggestion
- `out` parameters are not allowed in asynchronous methods.
- `out` parameters are not allowed in iterator methods.
- Properties cannot be passed as `out` parameters.

```

**Review by BillWagner — APPROVED** [review]
Thanks @cjvandyk 

I made a couple edits to match our style guide, and I'll :shipit:

**Comment by cjvandyk** [comment]
Thanks Bill!

**Pull request merged**